core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -594,9 +614,44 @@ private void setLocationOnce(String val, String qual, LocationType lt, boolean s
       }
       futureAndCurrentLocationSet = true;
     }
+    if (operationId != null) {
+      if (!suppressError) {
+        throw new IllegalStateException(
+            "Attempted to set location for tablet with an operation id. table ID: " + tableId
+                + " endrow: " + endRow + " -- operation id: " + operationId);
+      }
+    }
     location = new Location(val, qual, lt);
   }
 
+  /**
+   * Sets an operation ID only once.
+   *
+   * @param val operation id to set
+   * @param suppressError set to true to suppress an exception being thrown, else false
+   * @throws IllegalStateException if an operation id or location is already set
+   */
+  private void setOperationIdOnce(String val, boolean suppressError) {
+    // make sure there is not already an operation ID
+    if (operationId != null) {
+      if (!suppressError) {
+        throw new IllegalStateException(
+            "Attempted to set operation id for tablet that already has one. table ID: " + tableId
+                + " endrow: " + endRow + " -- operation id: " + operationId);
+      }
```

Comment:
Seems like the boolean should be set here.

```suggestion
      }
      operationIdAndCurrentLocationSet = true;
```

Reply:
Should it be set here in addition to where it is already set (within the if block below this)? Or should it only be set here?

Reply:
ignore this comment I made another comment that supersedes this one.